src/java.base/share/native/libjava/jni_util.c

```diff
@@ -534,7 +537,7 @@ static jstring
 newStringCp1252(JNIEnv *env, const char *str)
 {
     int len = (int) strlen(str);
-    jchar buf[512];
+    jchar buf[512] = {0};
```

Comment:
Also newSizedString8859_1, I would have expected it to complain there too.

Reply:
I think the compilers complain about `malloc'-ed paths, but we can harmonize `newSizedString8859_1` too. Please see new commit.